Xamarin.Forms.Platform.Android/Renderers/PageRenderer.cs

```diff
@@ -105,24 +104,27 @@ void UpdateHeight()
 			_previousHeight = newHeight;
 		}
 
-		void UpdateBackgroundColor(Page view)
+		void UpdateBackground()
 		{
-			if (view.Parent is BaseShellItem)
-			{
-				var background = view.BackgroundColor;
-				var color = Context.Resources.GetColor(global::Android.Resource.Color.BackgroundLight, Context.Theme);
-				SetBackgroundColor(background.IsDefault ? color : background.ToAndroid());
-			}
-			else if (view.BackgroundColor != Color.Default)
-				SetBackgroundColor(view.BackgroundColor.ToAndroid());
+			string bkgndImage = Element.BackgroundImage;
+			if (!string.IsNullOrEmpty(bkgndImage))
+				this.SetBackground(Context.GetDrawable(bkgndImage));
+			else
+				UpdatePageBackgroundColor();
 		}
 
-		void UpdateBackgroundImage(Page view)
+		void UpdatePageBackgroundColor()
 		{
-			if (!string.IsNullOrEmpty(view.BackgroundImage))
-				this.SetBackground(Context.GetDrawable(view.BackgroundImage));
+			if (Element.Parent is BaseShellItem)
+			{
+				Color background = Element.BackgroundColor;
+				if (background.IsDefault)
+					base.UpdateBackgroundColor();
+				else
+					SetBackgroundColor(Context.Resources.GetColor(global::Android.Resource.Color.BackgroundLight, Context.Theme));
+			}
```

Comment:
I am not too sure about the shell bits, but I think this has changed from

```csharp
if (shell) {
  if (isDefault) {
    SetBackground(GetFromTheme());
  } else {
    SetBackground(background.ToAndroid());
  }
}
```

to 

```csharp
if (shell) {
  if (isDefault) { // this has flipped
    base.Update();
  } else {
    SetBackground(GetFromTheme());
  }
}
```

You may be wanting:

```csharp
// a special case for shell
if (view.Parent is BaseShellItem && view.BackgroundColor == Color.Default) {
  var color = Context.Resources.GetColor(global::Android.Resource.Color.BackgroundLight, Context.Theme);
  SetBackgroundColor(color);
} else {
  UpdateBackgroundColor(); // we may not need the `base.` as we no longer have a method with the same name
}